opennms-doc/guide-admin/src/asciidoc/text/sentinel/sentinel.adoc

```diff
@@ -0,0 +1,305 @@
+In order to start features automatically, see chapter "Configuration" for more details.
+
+==== Karaf Extender
+The Karaf Extender was developed to make it easier to manage and extend the container using existing packaging tools.
+It allows packages to register Maven Repositories, Karaf Feature Repositories and Karaf Features to Boot by overlaying additional files, avoiding modifying any of the existing files.
+
+WARNING:    As the `karaf-extender` is at the moment used, it may be removed while developing sentinel further.
+
+=== Configuration
+This chapter describes the steps necessary to configure {opennms-product-name} with one _Sentinel_ and one _Minion_ to have the _Sentinel_ process flows.
+
+==== Sentinel
+It is assumed, that the _Sentinel_ container is running on a different system than the {opennms-product-name} and _Minion_.
+Therefore at least the following configurations are necessary:
+
+ - Configure the datasource to connect to the Postgres database
+ - Configure the controller (identity and connection to communicate with OpenNMS - same as for _Minion_)
+ - Configure the communication layer (for now either JMS or Kafka)
+ - Install features
+
+===== Configure the datasource
+
+```
+config:edit org.opennms.netmgt.distributed.datasource
+config:property-set datasource.url jdbc:postgresql://<db-host>:<db-port>/<db-name>
+config:property-set datasource.username <db-user>
+config:property-set datasource.password <db-password>
+config:property-set datasource.databaseName <db-name>
+config:update
+```
+
+NOTE:   TODO MVR provide a full list of supported properties, but for now
+        everything located in `features/distributed/datasource/src/main/resources/OSGI-INF/blueprint/blueprint.xml`
+        may be used for reference.
+
+==== Configure the controller
+
+```
+config:edit org.opennms.sentinel.controller
+config:property-set location SENTINEL <1>
+config:property-set id 00000000-0000-0000-0000-000000ddba11 <2>
+config:property-set http-url http://127.0.0.1:8980/opennms <3>
+config:property-set broker-url failover:tcp://127.0.0.1:61616 <4>
+config:update
+```
+<1> Not used at the moment
+<2> Must be unique overall _Sentinel_ and _Minion_ containers.
+<3> url which points to {opennms-product-name}
+<4> url which points to the {opennms-product-name} Active MQ Broker (for now this is always required)
+
+NOTE:   TODO MVR provide a full list of supported properties, but for now
+        everything located in `opennms/features/sentinel/core/src/main/resources/OSGI-INF/blueprint/blueprint.xml`
+        may be used for reference
+
+NOTE:   Basically the same properties as for the _Minion Controller_ are supported, but must be placed in a different file name.
+        `org.opennms.sentinel.controller.cfg` vs `org.opennms.minion.controller.cfg`.
+
+
+===== Install features
+
+The first step is to make the features known to the container:
+
+```
+repo-add mvn:org.apache.karaf.features/spring/4.1.5/xml/features
```

Comment:
Should we not include the feature repositories as part of the default configuration?

The set of features to install will depend on the particular deployment, but the repositories will always be needed.

Reply:
LoL, yes, you are right. I added these repositories probably 500 times during testing and didn't really question this anymore, until I read your comment and 🤦‍♂️ (-:

Reply:
Okay, after revisiting this, I now know why it is the way it is: There is no easy AND nice way to address this. 
The container is built as follows:
 * `container/shared` provides the basic distribution and is shared by minion and sentinel
 * `features/sentinel/repository` defines the system repository for the sentinel container
 * `features/container/sentinel` defines overrides of the shared container for sentinel and also builds the final distribution
 * `opennms-assemblies/sentinel` does some stuff on top of `features/container/sentinel`, which is not important here

The problem with this structure is, that the feature repositories are already defined in `container/shared` by the `karaf-maven-plugin`. However those must be overriden in module `features/container/sentinel`. But the `karaf-maven-plugin` does not support this functionality. 

Therefore one way to address this could be to place a `org.apache.karaf.features.cfg` file in `features/container/sentinel` containing the feature repositories for sentinel. But this means, any change to `container/shared` must be manually adapted to `features/container/sentinel`.

Another approach can be to NOT build a custom `system` directory, but the `default`repository and place everything in `repositories/default` instead of `system`. This allows us to use a `features.uri` file and place all custom repositories there. The `karaf-extender` takes care of the rest.

The last approach is similar to the previous one, except, we enhance the `karaf-extender` to load also a `features.uri` and `feature.boot` file from the system directory or from the `features.boot.d` directory. 

At the moment I am leaning forward the last approach, to enhance the `karaf-extender` to make use of the `features.uri` either in the `system` or `etc/features.boot.d` directory.

Reply:
I'd opt to just drop in a custom `org.apache.karaf.features.cfg` file and avoid using/enhancing the `karaf-extender`.